src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -322,22 +322,24 @@ public void move(Vector spdVec, double rot) {
    * @param rateLimit whether or not to use slew rate limiting
    */
   private void move(Vector spdVec, double rot, boolean rateLimit) {
+    Vector spdCommanded = spdVec.copy();
```

Comment:
Do not use copy here. Instead make a new private field called `m_spdCommanded` and copy the doubles into that vector. Copying effectively allocates a vector.

Reply:
but m_spdCommanded still needs to be initialized as as vector, so won't we allocate a vector either way?

Reply:
I mean you allocate only once in the method I suggest. In this case, you're just hiding "new Vector" inside the copy.